src/main/java/duke/program/TaskManager.java

```diff
@@ -0,0 +1,404 @@
+        taskCount++;
+    }
+
+    public void addNewEvent(String descriptionAndTime) throws EmptyDescriptionException, InvalidFormatException {
+
+        String[] descriptionAndAtTimeArray = descriptionAndTime.split(SEPARATOR_AT);
+        String description = descriptionAndAtTimeArray[0].trim();
+
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        if (descriptionAndAtTimeArray.length == 1) {
+            throw new InvalidFormatException();
+        }
+
+        String startAndEndTime = descriptionAndAtTimeArray[1].trim();
+
+        tasks.add(new Event(description, startAndEndTime));
+        taskCount++;
+    }
+
+    public void printAddedTaskMessage(Task task) {
+
+        String taskType = task.getType();
+        String taskDescription = task.getDescription();
+
+        System.out.println("Got it. I've added this task:");
+        if (taskType.equals(TASK_TYPE_ICON_TODO)) {
+            System.out.printf("[%s][ ] %s%n", taskType, taskDescription);
+        } else if (taskType.equals(TASK_TYPE_ICON_DEADLINE)) {
+            String byTime = task.getByDateTime();
+            System.out.printf("[%s][ ] %s(by:%s)%n", taskType, taskDescription, byTime);
+        } else if (taskType.equals((TASK_TYPE_ICON_EVENT))) {
+            String atTime = task.getStartAndEndTime();
+            System.out.printf("[%s][ ] %s(at:%s)%n", taskType, taskDescription, atTime);
+        } else {
+            printGenericErrorMessage();
+        }
+
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+    }
+
+    public void markAsDone(ArrayList<Task> tasks, String[] lineArgs) throws InvalidIndexException, TaskIndexOutOfBoundsException {
+
+        if (lineArgs.length > 2) {
+            throw new InvalidIndexException();
+        }
+
+        int doneIndex = Integer.parseInt(lineArgs[1]) - 1;
+
+        if (doneIndex >= taskCount || doneIndex < 0) {
+            throw new TaskIndexOutOfBoundsException();
+        } else {
+            if (tasks.get(doneIndex).getStatusIcon().equals(ICON_DONE)) {
+                printAlreadyDoneMessage();
+            } else {
+                tasks.get(doneIndex).setDone();
+                printMarkAsDoneMessage(tasks.get(doneIndex));
+            }
+        }
+    }
+
+    public void deleteTask(ArrayList<Task> tasks, String[] lineArgs) throws InvalidIndexException, TaskIndexOutOfBoundsException {
+
+        if (lineArgs.length > 2) {
+            throw new InvalidIndexException();
+        }
+
+        int deleteIndex = Integer.parseInt(lineArgs[1]) - 1;
+
+        if (deleteIndex >= taskCount || deleteIndex < 0) {
+            throw new TaskIndexOutOfBoundsException();
+        } else {
+            printDeletedTaskMessage(tasks.get(deleteIndex));
+            tasks.remove(deleteIndex);
+        }
+
+        taskCount--;
+    }
+
+    private void writeToFile(ArrayList<Task> tasks) throws IOException {
+
```

Comment:
There are generally few comments or method descriptors in your code. Provide method headers that describe non-obvious uses would be helpful - see [Java coding standard > comments](https://se-education.org/guides/conventions/java/basic.html#comments).